media-libs/libheif/libheif-1.7.0.ebuild

```diff
@@ -0,0 +1,71 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit autotools xdg-utils multilib-minimal
+
+if [[ ${PV} == "9999" ]] ; then
+	EGIT_REPO_URI="https://github.com/strukturag/${PN}.git"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/strukturag/${PN}/releases/download/v${PV}/${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+fi
+
+DESCRIPTION="ISO/IEC 23008-12:2017 HEIF file format decoder and encoder"
+HOMEPAGE="https://github.com/strukturag/libheif"
+
+LICENSE="GPL-3"
+SLOT="0/1.6"
+IUSE="static-libs test +threads"
+
+RESTRICT="!test? ( test )"
+
+BDEPEND="test? ( dev-lang/go )"
+DEPEND="
+	media-libs/libde265:=[${MULTILIB_USEDEP}]
+	media-libs/libpng:0=[${MULTILIB_USEDEP}]
+	media-libs/x265:=[${MULTILIB_USEDEP}]
+	sys-libs/zlib:=[${MULTILIB_USEDEP}]
+	virtual/jpeg:0=[${MULTILIB_USEDEP}]
+"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${P}-aom.patch"
```

Comment:
The build fails otherwise with:
```
heif_encoder_aom.cc:506:53: error: 'AOM_USAGE_REALTIME' was not declared in this scope
  506 |   unsigned int aomUsage = (encoder->realtime_mode ? AOM_USAGE_REALTIME : AOM_USAGE_GOOD_QUALITY);
      |                                                     ^~~~~~~~~~~~~~~~~~
heif_encoder_aom.cc:506:74: error: 'AOM_USAGE_GOOD_QUALITY' was not declared in this scope
  506 |   unsigned int aomUsage = (encoder->realtime_mode ? AOM_USAGE_REALTIME : AOM_USAGE_GOOD_QUALITY);
      |                                                                          ^~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/aom/aom_encoder.h:33,
                 from heif_encoder_aom.cc:36:
heif_encoder_aom.cc:543:40: error: 'AV1E_SET_ROW_MT' was not declared in this scope; did you mean 'AV1E_SET_MTU'?
  543 |     aom_codec_control(&encoder->codec, AV1E_SET_ROW_MT, 1);
      |                                        ^~~~~~~~~~~~~~~
heif_encoder_aom.cc:543:5: error: 'aom_codec_control_AV1E_SET_ROW_MT' was not declared in this scope; did you mean 'aom_codec_control_AV1E_SET_MTU'?
  543 |     aom_codec_control(&encoder->codec, AV1E_SET_ROW_MT, 1);
      |     ^~~~~~~~~~~~~~~~~
```
Patch is taken from https://github.com/strukturag/libheif/commit/6768552c0a99bb2957906be0f369850326486a58 and applied to this version.